server/channels/app/oembed/oembed.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package oembed
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+type OEmbedResponse struct {
+	// Type can be one of "photo", "video", "link", or "rich"
+	Type string `json:"type"`
+
+	// Fields that may be defined for any response type
+	Version         string `json:"version"`
+	Title           string `json:"title,omitempty"`
+	AuthorName      string `json:"author_name,omitempty"`
+	AuthorURL       string `json:"author_url,omitempty"`
+	ProviderName    string `json:"provider_name,omitempty"`
+	ProviderURL     string `json:"provider_url,omitempty"`
+	CacheAge        string `json:"cache_age,omitempty"`
+	ThumbnailURL    string `json:"thumbnail_url,omitempty"`
+	ThumbnailWidth  int    `json:"thumbnail_width,omitempty"`
+	ThumbnailHeight int    `json:"thumbnail_height,omitempty"`
+
+	// Fields that are required for responses with the type "photo"
+	URL string `json:"url"`
+
+	// Fields that are required for responses of the type "video" or "rich"
+	HTML string `json:"html"`
+
+	// Fields that are required for responses with the type "photo", "video", or "rich"
+	Width  int `json:"width"`
+	Height int `json:"height"`
+}
+
+func ResponseFromJson(r io.Reader) (*OEmbedResponse, error) {
```

Comment:
`ResponseFromJSON` would be more idiomatic.

Reply:
Good catch! I did my best to follow the Go convention here, but it's hard when JS uses the opposite